src/app/shared/infinite-scroll/infinite-scroll.directive.ts

```diff
@@ -0,0 +1,35 @@
+import {Directive, EventEmitter, HostListener, Input, Output} from '@angular/core';
+import {debounceTime, Subject} from 'rxjs';
+import {InfiniteScrollDirection} from './infinite-scroll-direction.enum';
+
+@Directive({
+    selector: '[appInfiniteScroll]',
+})
+export class InfiniteScrollDirective {
+    @Input() borderOffset: number;
+    @Input() delay: number;
+    @Output() loadData = new EventEmitter<InfiniteScrollDirection>();
```

Comment:
```suggestion
    @Output() loadData = this.scrollEventSubject.pipe(debounceTime(this.delay))
```

И подписка в конструкторе не нужна)